state/src/item/metadata.rs

```diff
@@ -93,28 +108,56 @@ const PREFIX: u8 = super::METADATA_PREFIX;
 
 impl Encodable for Metadata {
     fn rlp_append(&self, s: &mut RlpStream) {
+        let mut len = 4;
```

Comment:
Is this means `param_changed` implies they should fill the gap for `term` with a default value?
I think then this would can express the intention more.
```rust
const INITIAL_LEN: usize = 4;
const TERM_LEN: usize = 2;
const PARAMS_LEN: usize = 2;
let mut len = INITIAL_LEN;
if term_changed || params_chagned { 
    // Fill default values for term if params_changed even if !term_changed
    len += TERM_LEN;
}
if params_chagned {
    len += PARAMS_LEN;
}
```
and

```rust
s.begin_list(len)
    .append(&PREFIX)
    .append(&self.number_of_shards)
    .append(&self.number_of_initial_shards)
    .append_list(&self.hashes);
if term_changed {
    // append term
} else if params_changed {
    // append default value for term
}
if params_changed {
    // append params
}
```

Reply:
@foriequal0 fixed

Reply:
I was pointing out the control flow, not only labeling the constant. However, labeling makes it much better than before.